src/components/Data/Collections/Watch.vue

```diff
@@ -49,51 +129,275 @@
+        basicFilter,
+        basicFilterForm
+      }
+    },
     data () {
       return {
         subscribed: false,
-        room: null
+        room: null,
+        filters: {},
+        availableFilters,
+        formatFromBasicSearch,
+        subscribeOptions: {scope: 'all', users: 'all', state: 'all'},
+        notifications: [],
+        notificationsLengthLimit: 50,
+        warning: {message: '', count: 0, lastTime: null, info: false},
+        scrollGlueActive: true,
+        scrollListener: null
+      }
+    },
+    watch: {
+      index () {
+        this.reset()
+      },
+      collection () {
+        this.reset()
+      }
+    },
+    ready () {
+      // display the toolbar when user scroll (or when scroll glue is active)
+      let scrolled = false
+      let toolbar = document.getElementById('notification-controls-fixed')
+
+      this.notifications = []
+
+      window.onscroll = function () {
+        scrolled = true
+      }
+
+      // delay position checking in an interval instead of onscroll event to reduce lags
+      this.scrollListener = setInterval(function () {
+        if (scrolled) {
+          scrolled = false
+
+          if (window.scrollY > 200 && toolbar.classList.contains('closed')) {
+            toolbar.classList.remove('closed')
+          } else if (window.scrollY <= 200 && !toolbar.classList.contains('closed')) {
+            toolbar.classList.add('closed')
+          }
+        }
+      }, 100)
+    },
+    destroyed () {
+      // trigged when user leave watch data page
+      if (this.scrollListener !== null) {
+        clearInterval(this.scrollListener)
       }
+      this.reset()
     },
     directives: [
-      jQueryCollapsible
+      jQueryCollapsible,
+      JsonFormatter,
+      ScrollGlue
     ],
     components: {
       Notification,
       CollectionDropdown,
+      SubscriptionControls,
+      Filters,
       Headline
     },
     methods: {
-      manageSub (index, collection) {
+      basicSearch (filters) {
+        if (!filters) {
+          this.$router.go({query: {basicFilter: ''}})
+          return
+        }
+
+        let basicFilter = JSON.stringify(filters)
+        this.$router.go({query: {basicFilter}})
+      },
+      rawSearch (filters) {
+        if (!filters) {
+          this.$router.go({query: {rawFilter: ''}})
+          return
+        }
+
+        let rawFilter = JSON.stringify(filters)
+        this.$router.go({query: {rawFilter}})
+      },
+      refreshSearch () {
+        this.$router.go({query: {...this.$route.query}})
+      },
+      setScrollGlue (value) {
+        this.scrollGlueActive = value
+      },
+      toggleSubscription () {
         if (!this.subscribed) {
           this.subscribed = true
-          this.room = this.subscribe(index, collection)
+          this.room = this.subscribe(this.filters, this.index, this.collection)
         } else {
+          this.reset()
+        }
+      },
+      notificationToMessage (notification) {
```

Comment:
Why did you put this in the component? I really think this should be a private utility function.